tools/check_wav_file_snr.m

```diff
@@ -0,0 +1,355 @@
+meas.snr_db = signal_db - noise_db;
+
+i1 = find(t > param.t_ignore_from_start, 1, 'first');
+i2 = find(t <= t(end) - param.t_ignore_from_end, 1, 'last');
+
+meas.mean_signal_db = mean(meas.signal_db(i1 :i2));
+meas.mean_noise_db = mean(meas.noise_db(i1 :i2));
+meas.mean_snr_db = mean(meas.snr_db(i1 :i2));
+meas.max_noise_db = max(meas.noise_db(i1 :i2));
+meas.min_snr_db = min(meas.snr_db(i1 :i2));
+
+end
+
+function meas = check_tone_levels(stft, param, meas, t)
+
+meas.t_glitch = 0;
+meas.num_glitches = 0;
+meas.success = true;
+
+% Find glitches from SNR curve drops
+i1 = find(t > param.t_ignore_from_start, 1, 'first');
+i2 = find(t <= t(end) - param.t_ignore_from_end, 1, 'last');
+idx = find(meas.snr_db(i1:i2) < meas.mean_snr_db - param.max_snr_drop);
+
+if ~isempty(idx)
+	idx = idx + i1 - 1;
+	didx = diff(idx);
+	meas.num_glitches = 1 + length(find(didx > 2));
+	start_idx = idx(1);
+	cidx = find(didx(1:end) > 1, 1, 'first');
+	if isempty(cidx)
+		end_idx = idx(end);
+	else
+		end_idx = idx(cidx);
+	end
+	meas.t_glitch = param.t_step * mean([start_idx end_idx] - 1) + ...
+		0.5 * param.n_fft / param.fs;
+	meas.success = false;
+end
+
+if meas.min_snr_db < param.min_snr
+	meas.success = false;
+end
+
+end
+
+function [signal_idx, tone_f] = find_test_tone(stft, param, f, t, test_tone_f)
+
+if test_tone_f > 0
+	err_ms = (f - test_tone_f) .^ 2;
+	signal_idx = find(err_ms == min(err_ms));
+	tone_f = f(signal_idx);
+	return
+end
+
+i1 = find(t > param.t_ignore_from_start, 1, 'first');
+i2 = find(t <= t(end) - param.t_ignore_from_end, 1, 'last');
+
+signal_idx_all = zeros(i2 - i1 + 1, 1);
+for i = i1 : i2
+	signal_idx_all(i - i1 + 1) = find(stft(:, i) == max(stft(:, i)),1, 'first');
+end
+
+signal_idx = round(mean(signal_idx_all));
+tone_f = f(signal_idx);
+
+end
+
+function [x, fs, frames, channels] = read_audio(fn)
+
+[x, fs] = audioread(fn);
+sx = size(x);
+frames = sx(1);
+channels = sx(2);
+
+end
+
+function [stft, f, t, param] = compute_stft(x, fs, param)
+
+sx = size(x);
+if sx(2) > 1
+	error('One channel only');
+end
+
+frames = sx(1);
+win = kaiser(param.n_fft, 20);
+param.win_spread = 7;
+param.win_gain = -13.0379;
+param.fs = fs;
+
+param.n_step = fs * param.t_step;
+param.n_stft = floor((frames - param.n_fft) / param.n_step);
+n_half_fft = param.n_fft / 2 + 1;
+scale = 1 / param.n_fft;
+f = linspace(0, fs/2, n_half_fft);
+t = (0 : (param.n_stft - 1)) * param.t_step;
+stft = zeros(n_half_fft, param.n_stft);
+
+for i = 1 : param.n_stft
+	i1 = (i - 1) * param.n_step + 1;
+	i2 = i1 + param.n_fft - 1;
+	s1 = fft(x(i1 : i2) .* win) * scale;
+	s2 = s1(1 : n_half_fft);
+	s = s2 .* conj(s2);
+	stft(:, i) = s;
+end
+
+end
+
+function fh = plot_specgram(stft, f, t, tstr)
+
+fh = figure;
+clims = [-160 0];
+imagesc(1e3 * t, f, 10*log10(stft), clims)
+set(gca, 'ydir', 'normal');
+colorbar;
+grid on;
+title(tstr, 'interpreter', 'none');
+xlabel('Time (ms)');
+ylabel('Frequency (Hz)');
+
+end
+
+function fh = plot_levels(meas, t, idstr)
```

Comment:
Need to find a way to avoid these plots to appear as ASCII to log file. Just PNG file export to log dir is enough.